external-crates/move/crates/move-binary-format/src/compatibility.rs

```diff
@@ -92,15 +275,25 @@ impl Compatibility {
 
     /// Check compatibility for `new_module` relative to old module `old_module`.
     pub fn check(&self, old_module: &Module, new_module: &Module) -> PartialVMResult<()> {
-        let mut datatype_and_function_linking = true;
-        let mut datatype_layout = true;
-        let mut friend_linking = true;
-        let mut entry_linking = true;
-        let mut no_new_variants = true;
+        self.check_::<ExecutionCompatibilityMode>(old_module, new_module)
+            .map_err(|_| PartialVMError::new(StatusCode::BACKWARD_INCOMPATIBLE_MODULE_UPDATE))
+    }
+
+    pub fn check_<M: CompatibilityMode>(
```

Comment:
nit: we should probably pick a more information name for this now -- how about `check_with_mode` or something?

Reply:
I actually meant to ask about this because I wasn't liking the name either.